modules/shared/src/main/scala/io/constellationnetwork/currency/http/Decoders.scala

```diff
@@ -0,0 +1,42 @@
+package io.constellationnetwork.currency.http
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+
+import io.constellationnetwork.currency.dataApplication.DataTransaction.DataTransactions
+import io.constellationnetwork.currency.dataApplication.{DataTransaction, DataUpdate, FeeTransaction}
+import io.constellationnetwork.security.signature.Signed
+
+import derevo.circe.magnolia.decoder
+import derevo.derive
+import io.circe.Decoder
+import org.http4s.circe.jsonOf
+import org.http4s.{DecodeResult, EntityDecoder, InvalidMessageBodyFailure}
+
+object Decoders {
+
+  def feeTransactionRequestDecoder[F[_]: Async, D <: DataUpdate](
+    implicit dataUpdateDecoder: Decoder[D],
+    feeTransactionDecoder: Decoder[FeeTransaction]
+  ): EntityDecoder[F, DataTransactions] = {
+    implicit val signedDataUpdateDecoder: Decoder[Signed[D]] = Signed.decoder[D]
+    implicit val signedFeeTransactionDecoder: Decoder[Signed[FeeTransaction]] = Signed.decoder[FeeTransaction](feeTransactionDecoder)
+
+    @derive(decoder)
+    case class DataTransactionRequest(
+      data: Signed[D],
+      fee: Option[Signed[FeeTransaction]]
+    )
+
+    jsonOf[F, DataTransactionRequest].flatMapR { req =>
+      val signedTransactions: List[Signed[DataTransaction]] = List(req.data) ++ req.fee.toList
```

Comment:
`List(req.data, req.fee)` (I think I've already made that comment 🤔 )

Reply:
It not works:
Type mismatch. Required: List[Signed[DataTransaction]], found: List[Product with Serializable]
I think it's due req.fee being option

Reply:
ah, ok :+1: